[PE-D][Tester D] Corrupted data files causes total system failure

**Defect**
_NOTE: this defect is distinct from a very similarly named defect. This defect is **SIGNIFICANTLY WORSE**._

When a data file is corrupted in a very particular manner, attempts to start the program will cause an instant crash and exit without any indication to the user.

_Reproduction_
1. Start with the base data set.
2. Add a feeding session: `feed f/Alex Yeoh n/Max dt/2020-10-10 10:00`
3. Close the app.
4. Edit `data/addressbook.json` in this manner: change the feeding session `dateTime` field to `0000` (or any invalid input).

Actual behaviour: instant system crash (silent; only logged in console).

```
Exception in Application init method
Exception in thread "main" java.lang.RuntimeException: Exception in Application init method
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:888)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.time.format.DateTimeParseException: Text '0000' could not be parsed at index 4
        at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2052)
        at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1954)
        at java.base/java.time.LocalDateTime.parse(LocalDateTime.java:494)
        at seedu.address.storage.JsonAdaptedFeedingSession.toModelType(JsonAdaptedFeedingSession.java:69)
        at seedu.address.storage.JsonSerializableAddressBook.toModelType(JsonSerializableAddressBook.java:70)
        at seedu.address.storage.JsonAddressBookStorage.readAddressBook(JsonAddressBookStorage.java:55)
        at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:64)
        at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:58)
        at seedu.address.MainApp.initModelManager(MainApp.java:82)
        at seedu.address.MainApp.init(MainApp.java:63)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:817)
        ... 2 more
```

**Program will refuse to start on any further attempt unless the data file is completely deleted or the error is fixed manually.**

Expected: Any sane error handling/attempt at recovery for corrupt data files.

_Severity Justification_
High. Very high hindrance: unable to use the app at all on this failure. User has to manually debug this.

**Suggestion**
As an advertised feature (editing data file), some graceful error handling is expected. Probably just a missed edge case, since other failed edits still allow the program to start up.

-------------
Labels: `severity.High` `type.FunctionalityBug`

--- a/src/main/java/seedu/address/storage/JsonAdaptedFeedingSession.java
+++ b/src/main/java/seedu/address/storage/JsonAdaptedFeedingSession.java
@@ -2,6 +2,7 @@ package seedu.address.storage;
 
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.UUID;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
@@ -69,6 +70,8 @@ class JsonAdaptedFeedingSession {
             modelDateTime = LocalDateTime.parse(dateTime, DATE_TIME_FORMATTER);
         } catch (IllegalArgumentException e) {
             throw new IllegalValueException("Invalid UUID or DateTime format");
+        } catch (DateTimeParseException e) {
+            throw new IllegalValueException("Corrupted DateTime field: " + dateTime);
         }
 
         final String modelNotes = notes != null ? notes : "";

